### PR TITLE
Add IntegrationTests for chat2 API

### PR DESCRIPTION
**Pull request by vanitasvitae**
* Test if IncomingMessageListeners are getting fired for incoming chat messages.
* Test if OutgoingMessageListeners are getting fired for outgoing chat messages.

### CODE REVIEW

File: smack-integration-test/src/main/java/org/jivesoftware/smackx/chat/IncomingMessageListenerIntegrationTest.java [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+import org.junit.After;
+import org.jxmpp.jid.EntityBareJid;
+
+public class IncomingMessageListenerIntegrationTest extends AbstractSmackIntegrationTest {
+
+    public IncomingMessageListenerIntegrationTest(SmackIntegrationTestEnvironment environment) {
+        super(environment);
+    }
+
+    private final String body = StringUtils.randomString(16);
```
Comment (Flowdalic) [inline smack-integration-test/src/main/java/org/jivesoftware/smackx/chat/IncomingMessageListenerIntegrationTest.java]: Those three should all be local variables in the test method.

File: smack-integration-test/src/main/java/org/jivesoftware/smackx/chat/IncomingMessageListenerIntegrationTest.java [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+        ChatManager m2 = ChatManager.getInstanceFor(conTwo);
+
+        m2.addIncomingListener(listener);
+        Chat chat = m1.chatWith(conTwo.getUser().asEntityBareJid());
+        chat.send(body);
+        syncPoint.waitForResult(10 * 1000);
+    }
+
+    @After
+    public void cleanup() {
```
Comment (Flowdalic) [inline smack-integration-test/src/main/java/org/jivesoftware/smackx/chat/IncomingMessageListenerIntegrationTest.java]: Don't use an @After method for this, instead use a `finally` clause within the test method.

File: smack-integration-test/src/main/java/org/jivesoftware/smackx/chat/IncomingMessageListenerIntegrationTest.java [hunk 3]
```diff
@@ -0,0 +1,64 @@
...
+        public void newIncomingMessage(EntityBareJid from, Message message, Chat chat) {
+            if (body.equals(message.getBody())) {
+                syncPoint.signal();
+            }
+        }
+    };
+
+    @SmackIntegrationTest
+    public void test() throws Exception {
+        ChatManager m1 = ChatManager.getInstanceFor(conOne);
```
Comment (Flowdalic) [inline smack-integration-test/src/main/java/org/jivesoftware/smackx/chat/IncomingMessageListenerIntegrationTest.java]: The ChatManagers should be fields of the test class. And they should be named after the patter the other tests use, i.e. `chatManagerOne` and `chatManagerTwo`.

**Comment by vanitasvitae** [comment]
Squashed.

**Pull request merged**